terraform/modules/front_door/waf.tf

```diff
@@ -0,0 +1,164 @@
+provider "aws" {
+  alias  = "north-virginia"
+  region = "us-east-1"
+
+  assume_role {
+    role_arn = var.provider_role_arn
+  }
+}
+
+resource "aws_wafv2_web_acl" "this" {
+  #checkov:skip=CKV2_AWS_31:TODO CLDC-2781 setup WAF logging in cloudwatch
+  name        = var.prefix
+  description = "Web ACL to restrict traffic to CloudFront"
+  provider    = aws.north-virginia
+  scope       = "CLOUDFRONT"
+
+  default_action {
+    allow {}
+  }
+
+  visibility_config {
+    cloudwatch_metrics_enabled = true
+    metric_name                = "waf-metrics"
+    sampled_requests_enabled   = false
+  }
+
+  rule {
+    name     = "AWSManagedRulesAmazonIpReputationList"
+    priority = 1
+
+    override_action {
+      none {}
+    }
+
+    statement {
+      managed_rule_group_statement {
+        name        = "AWSManagedRulesAmazonIpReputationList"
+        vendor_name = "AWS"
+      }
+    }
+
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "waf-block-disreputable-ip-metrics"
+      sampled_requests_enabled   = true

```

Comment:
For all of our WAF rules, I thought it would be good to have sample requests enabled if a bad request is ever trapped by a rule. I believe these are free according to the console , see screenshot:
![image](https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data-infrastructure/assets/109098469/9cd8e76c-bb8b-4f9b-aa82-41f0407d3f3e)